### ISSUE TITLE
Should AuthSigninMethod use Post?

### ISSUE THREAD

**Issue by ivan-m** [NCrashed/servant-auth-token-api#1]
Currently, there is:

```haskell
type AuthSigninMethod 
    = "auth" :> "signin" 
        :> QueryParam "login" Login 
        :> QueryParam "password" Password 
        :> QueryParam "expire" Seconds
        :> Get '[JSON] (OnlyField "token" SimpleToken)
```

However, this means that any parameters will appear in the resulting URL.  As such, they can appear in server logs, etc.

If, however, `Post` is used (as is the case with `AuthSigninPostCodeMethod`) then these parameters will not appear in logs, usernames and passwords can be longer, and if for some reason someone tries to call one in a browser then credentials won't be found in the history and the call won't be cached.

As another point, `Get` is typically used for idempotent requests, whereas each time you sign in you should expect a new token; as such, from a conceptual point of view I don't think it aligns well anyway.

**Comment by NCrashed** [comment]
Yep, I think that `Post` method is more applicable here. But we need to retain  `Get` version for some time as deprecated.

**Comment by ivan-m** [comment]
Note that the use of `QueryParam` in `AuthSigninPostCodeMethod` is probably also a potential security hole (a bit more arguable since the code isn't re-usable though).

What probably needs to happen is to have a new datatype for `AuthSigninPostMethod` to accept in a `ReqBody` (which can then take `JSON`, `FormUrlEncoded`, etc.).  However, because of this `authSigninMethod` from the server can't be simply re-used as the types would no longer match.

**Comment by NCrashed** [comment]
Sorry for such long delay. I will add the change with JSON body for `AuthSigninPostMethod` to your original changes.

**Comment by NCrashed** [comment]
Note: publish release in sync with server package.

**Comment by hughjfchen** [comment]
The authSigninPost handler function merged, but not export from the module list. Please refer to this PR to export it. #5
